igor_src/main.cpp

```diff
@@ -1876,8 +1954,193 @@ int main(int argc , char* argv[]){
 					return terminate_IGoR_with_error_message("Exception caught upon aligning J genomic templates.",e);
 				}
 			}
+			
+			//Get CDR3 from alignments.
+			if(align_v and align_j){
```

Comment:
This condition should not be necessary ! The only necessary condition is to have already run alignments and have the files present at the right place with the right names.
By putting this condition you enforce the user to extract CDR3s in the same run than the alignment run.
This would make sense if you were extracting CDR3s on the fly of the alignment procedure (the same way the CDR3s can be extracted on the fly upon generating synthetic sequences).

Reply:
Ok this is a function location problem because I need those files after the alignments. but IGoR can create the alignment files separately. So that means I will need to make a new command in IGoR to create the CDR3 files. Something like $ igor -extractFeatures --CDR3 ...
So I started to overthink on this and at the end, A&T told me to put it as default in the alignments. So I know is bad, but I didn't have a better idea to put it. So if you have a better suggestion to do this will be great ;)

Reply:
Ok I see your point. I was also thinking making a new command or a new `-align` subcommand.
Here are the pros and cons I see at making a new command:
Pros:
- this is the way MiXCR works (first produces a .vdjca file with alignments then use the -export command, see https://mixcr.readthedocs.io/en/master/align.html). I think this workflow works great, see my second and 3rd points.
- alignments are computationally intensive (at least in IGoR) and in case the user forgets to add the CDR3 flag and extraction is made at align time he has to rerun the alignment.
- alignments take a lot of disk space and we should aim at compressing them, I think this is the reason why MiXCR uses .vdjca files, they take up much less space than plain text. Adding the CDR3 information by default as plain text would also take even more disk space
- we should aim at conforming with AIRR data standard (see #18 ) both for alignments and probabilistic assignments. Producing the AIRR data requires to aggregate assignments for a given sequence if I recall correctly

Cons: 
- we should figure out how the equivalent feature could be added in the probabilistic framework, commands should be consistent between the two. For the probabilistic framework this would mean being able to post process the best scenario file (which is not produced by default). For now I would say that the `-output` command is the equivalent feature, and the result is given on the fly (that being said I now realize that `-output` should probably be refactored as `--output`, meaning as a suboption of `-infer` or `-evaluate`).